lib/robot.js

```diff
@@ -1,28 +1,35 @@
-const {EventEmitter} = require('promise-events')
 const GitHubApi = require('github')
 const Bottleneck = require('bottleneck')
 const express = require('express')
-const Context = require('./context')
 
 /**
  * The `robot` parameter available to apps
  *
  * @property {logger} log - A logger
  */
 class Robot {
-  constructor ({app, cache, logger, router, catchErrors} = {}) {
-    this.events = new EventEmitter()
+  constructor ({app, cache, logger, webhook = {}, router, catchErrors} = {}) {
     this.app = app
     this.cache = cache
     this.router = router || new express.Router()
     this.log = wrapLogger(logger)
     this.catchErrors = catchErrors
+    this.webhook = webhook
+    this.on = webhook.on
+    this.removeListener = webhook.removeListener
   }
 
   async receive (event) {
-    return this.events.emit('*', event).then(() => {
-      return this.events.emit(event.event, event)
-    })
+    if (event.event && event.payload) {
+      console.log('DEPRECATED: robot.receive({event, payload}) is now robot.receive({id, name, data})')
+      event = {
+        id: 'unknown',
+        name: event.event,
+        data: event.payload
+      }
+    }
+
+    return this.webhook.receive(event)
   }
```

Comment:
before, each `robot` instance had their own state for events. Now the `probot` process has a singleton `webhook` API that it passes to all robot instances. That means that if one robot receives a "push" event, all robot instances receive a push event. I assume this is the correct behavior and the way it worked before, too, but now the implementation is different.

We could in theory just get rid of the `robot.receive` method, because calling the upper level `probot.receive` method will have the same effect.

The alternative would be for all robot instances to have their own `webhook` instance, but I don’t think this is what we want?

I’m honestly a bit confused by the whole `robots[]` array :)

Reply:
the current implementation breaks some tests of existing probot apps, because a `webhooks` must now be passed to `require('probot').createRobot`. Before the method could be called without any option, like here: https://github.com/probot/settings/blob/30caca9a6ec3834ffaf606c7d5f96399c20102e7/test/index.test.js#L10

Reply:
Had a call with Brandon. For now we want robot instances to work in isolation, so I’ll give them separate EventHandler instances similar to how it worked before